### PR TITLE
[TGBOT-297] Create HttpClient EngineerGateway implementation

### PR DESCRIPTION
**Pull request by AlexanderYarokhovich**
https://schneidervs.atlassian.net/jira/software/projects/TGBOT/boards/2?selectedIssue=TGBOT-297

### CODE REVIEW

**Review by Sav4ykVladzislau — DISMISSED** [review]

**Review by yurasava — DISMISSED** [review]

**Review by Artem-Metlushko — DISMISSED** [review]

File: http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+        sendHttpRequest(request);
+        return engineer;
+    }
+
+    private void validateEngineer(Engineer engineer) {
+        if (engineer == null) {
+            throw new IllegalArgumentException("Engineer can't be null");
+        }
+
```
Comment (max-java) [inline http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java]: validation should be moved to separate class. 
you can create validator functional interface, and then autowire all validators in Collestion in service, and then just go one by one. 

So validation would be a separate story, please create jira ticket

File: http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            return httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException | InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException("Can't read getStringHttpResponse", e);
+        }
+    }
+
+    private void checkResponse(HttpResponse<String> response) {
+        if (response.statusCode() < 200 || response.statusCode() > 299) {
+            throw new RuntimeException("Can't to save engineer: " + response.body());
```
Comment (max-java) [inline http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java]: please double check error message

File: http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        }
+    }
+
+    @Override
+    public boolean deleteEngineerById(String engineerId) {
+        return false;
```
Comment (max-java) [inline http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java]: throw NotImplementedException

**Review by max-java — CHANGES_REQUESTED** [review]
ptal

File: http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java (hunk 1 above)
Comment (AlexanderYarokhovich) [reply http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java]: done

File: http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java (hunk 2 above)
Comment (AlexanderYarokhovich) [reply http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java]: done

File: http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java (hunk 3 above)
Comment (AlexanderYarokhovich) [reply http-client/src/main/java/com/tutrit/httpclient/HttpEngineerGateway.java]: done

**Review by yurasava — APPROVED** [review]

**Review by Sav4ykVladzislau — APPROVED** [review]

**Review by Artem-Metlushko — APPROVED** [review]

**Review by schneidervs — APPROVED** [review]

**Review by max-java — APPROVED** [review]
lgfm

**Pull request merged**